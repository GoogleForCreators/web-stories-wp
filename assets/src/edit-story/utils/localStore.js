/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { trackError } from '@web-stories-wp/tracking';

export const LOCAL_STORAGE_PREFIX = {
  PANEL: 'web_stories_ui_panel_settings',
  TEXT_SET_SETTINGS: 'web_stores_text_set_settings',
  TERMS_MEDIA3P: 'web_stories_media3p_terms_agreement',
  VIDEO_OPTIMIZATION_DIALOG_DISMISSED:
    'web_stories_video_optimization_dialog_dismissed',
  HELP_CENTER: 'web_stories_help_center',
};

function getItemByKey(key) {
  let parsed = null;
  try {
    const stored = localStorage.getItem(key);
    parsed = JSON.parse(stored);
  } catch (e) {
    trackError('local_storage_read', e.message);
  }
  return parsed;
}

function setItemByKey(key, data) {
  try {
    localStorage.setItem(key, JSON.stringify(data));
  } catch (e) {
    trackError('local_storage_write', e.message);
  }
}

export default {
  getItemByKey,
  setItemByKey,
};
