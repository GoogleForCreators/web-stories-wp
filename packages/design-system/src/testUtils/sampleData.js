/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const basicDropDownOptions = [
  {
    label: 'label item one',
    value: 'label-item-one',
  },
  {
    label: 'label item two',
    value: 'label-item-two',
  },
  {
    label: 'label item three',
    value: 'label-item-three',
  },
  {
    label: 'label item four (disabled)',
    value: 'label-item-four',
    disabled: true,
  },
  {
    label: 'label item five (value is a number)',
    value: 5,
  },
  {
    label: 'label item six (value is a boolean)',
    value: true,
  },
  {
    label: 'drafts',
    value: 'drafts',
  },
  {
    label: 'cats',
    value: 'cats',
  },
  {
    label: 'dogs',
    value: 'dogs',
  },
  {
    label: 'parakeets',
    value: 'parakeets',
  },
  {
    label: 'lemurs',
    value: 'lemurs',
  },
  {
    label: 'ocelots',
    value: 'ocelots',
  },
];

export const effectChooserOptions = [
  {
    value: 'none',
    label: 'none',
    width: 'full',
  },
  {
    value: 'drop in',
    label: 'drop',
    width: 'full',
  },
  {
    value: 'fly in left-to-right',
    label: 'fly in',
    width: 'half',
  },
  {
    value: 'fly in top-to-bottom',
    label: 'fly in',
    width: 'half',
  },
  {
    value: 'fly in right-to-left',
    label: 'fly in',
    width: 'half',
  },
  {
    value: 'fly in bottom-to-top',
    label: 'fly in',
    width: 'half',
  },
  {
    value: 'pulse',
    label: 'pulse',
    width: 'full',
  },
  {
    value: 'rotate in left-to-right',
    label: 'rotate in',
    width: 'half',
  },
  {
    value: 'rotate in right-to-left',
    label: 'rotate in',
    width: 'half',
  },
  {
    value: 'twirl',
    label: 'twirl',
    width: 'full',
  },
];

export const nestedDropDownOptions = [
  {
    label: 'aliens',
    options: [
      { value: 'alien-1', label: 'ET' },
      { value: 'alien-2', label: 'Stitch' },
      { value: 'alien-3', label: 'Groot' },
      { value: 'alien-4', label: 'The Worm Guys' },
      { value: 'alien-5', label: "Na'vi" },
      { value: 'alien-6', label: 'Arachnids' },
      { value: 'alien-7', label: 'The Predator' },
      { value: 'alien-8', label: 'Xenomorph' },
    ],
  },
  {
    label: 'dragons',
    options: [
      { value: 'dragon-1', label: 'Smaug' },
      { value: 'dragon-2', label: 'Mushu' },
      { value: 'dragon-3', label: 'Toothless' },
      { value: 'dragon-4', label: 'Falkor' },
      { value: 'dragon-5', label: 'Drogon' },
      { value: 'dragon-6', label: 'Kalessin' },
    ],
  },
  {
    label: 'dogs',
    options: [
      { value: 'dog-1', label: 'Snoopy' },
      { value: 'dog-2', label: 'Scooby' },
    ],
  },
];

export const badOptions = [
  { value: 0, label: '0 as a number' },
  { value: '0', label: '0 as a string' },
  { value: false, label: 'false as a boolean' },
  { value: 'false', label: 'false as a string' },
  { value: true, label: 'true as a boolean' },
  { value: undefined, label: "undefined and shouldn't come through" },
];
