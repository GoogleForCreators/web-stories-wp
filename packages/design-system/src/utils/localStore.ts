/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { trackError } from '@googleforcreators/tracking';

export const LOCAL_STORAGE_PREFIX = {
  PANEL: 'web_stories_ui_panel_settings',
  TEXT_SET_SETTINGS: 'web_stores_text_set_settings',
  TERMS_MEDIA3P: 'web_stories_media3p_terms_agreement',
  VIDEO_OPTIMIZATION_DIALOG_DISMISSED:
    'web_stories_video_optimization_dialog_dismissed',
  HELP_CENTER: 'web_stories_help_center',
  DELETE_COLOR_PRESET_DIALOG_DISMISSED:
    'web_stories_delete_color_preset_dialog_dismissed',
  DELETE_STYLE_PRESET_DIALOG_DISMISSED:
    'web_stories_delete_style_preset_dialog_dismissed',
  DEFAULT_VIEW_PAGE_TEMPLATE_LAYOUT: 'web_stories_default_template_view',
  BACKGROUND_IS_SET_DIALOG_DISMISSED:
    'web_stories_background_is_set_dialog_dismissed',
  CORS_CHECK_DIALOG_DISMISSED: 'web_stories_cors_check_dialog_dismissed',
  MEDIA_RECORDING_AUDIO_INPUT: 'web_stories_media_recording_audio_input',
  MEDIA_RECORDING_VIDEO_INPUT: 'web_stories_media_recording_video_input',
  ELEMENT_TOOLBAR_SETTINGS: 'web_stories_toolbar_settings',
  MEDIA_RECORDING_VIDEO_EFFECT: 'web_stories_media_recording_video_effect',
};

function getItemByKey(key: string): unknown {
  let parsed: unknown = null;
  try {
    const stored = localStorage.getItem(key);
    parsed = stored !== null ? JSON.parse(stored) : stored;
  } catch (err) {
    if (err instanceof Error) {
      void trackError('local_storage_read', err.message);
    }
  }
  return parsed;
}

function setItemByKey(key: string, data: unknown) {
  try {
    localStorage.setItem(key, JSON.stringify(data));
  } catch (err) {
    if (err instanceof Error) {
      void trackError('local_storage_write', err.message);
    }
  }
}

export const localStore = {
  getItemByKey,
  setItemByKey,
};
