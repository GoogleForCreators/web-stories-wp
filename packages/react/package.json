{
  "name": "@googleforcreators/react",
  "description": "A tiny wrapper package that re-exports React and provides additional utilities and custom hooks.",
  "private": false,
  "version": "0.1.202204141511",
  "author": {
    "name": "Google",
    "url": "https://creators.google/"
  },
  "license": "Apache-2.0",
  "keywords": [
    "web stories"
  ],
  "homepage": "https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/react/README.md",
  "repository": {
    "type": "git",
    "url": "https://github.com/GoogleForCreators/web-stories-wp.git",
    "directory": "packages/react"
  },
  "bugs": {
    "url": "https://github.com/googleforcreators/web-stories-wp/issues"
  },
  "engines": {
    "node": ">= 12 || >= 14 || >= 16",
    "npm": ">= 7.3"
  },
  "customExports": {
    ".": {
      "default": "./src/index.js"
    }
  },
  "main": "dist/index.js",
  "module": "dist-module/index.js",
  "source": "src/index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "prop-types": "^15.8.1",
    "resize-observer-polyfill": "^1.5.1",
    "shallow-equal": "^1.2.1",
    "use-context-selector": "^1.2.11",
    "use-debounce": "^7.0.1"
  },
  "devDependencies": {
    "@testing-library/react": "^12.1.5",
    "@testing-library/react-hooks": "^8.0.0"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
