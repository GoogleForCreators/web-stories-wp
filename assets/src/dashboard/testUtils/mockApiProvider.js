/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { useMemo, useState } from 'react';
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */

import { ApiContext } from '../app/api/apiProvider';

const noop = () => {};

export default function MockApiProvider({ children, value }) {
  const [currentUser, setCurrentUser] = useState(getCurrentUserState());

  const usersApi = useMemo(
    () => ({
      fetchCurrentUser: noop,
      toggleWebStoriesTrackingOptIn: () =>
        setCurrentUser(toggleOptInTracking(currentUser)),
    }),
    [currentUser]
  );

  const mergedValue = useMemo(
    () => ({
      state: { currentUser },
      actions: { usersApi },
      ...value,
    }),
    [currentUser, usersApi, value]
  );
  return (
    <ApiContext.Provider value={mergedValue}>{children}</ApiContext.Provider>
  );
}

MockApiProvider.propTypes = {
  children: PropTypes.node,
  value: PropTypes.object,
};

function getCurrentUserState() {
  return {
    data: { id: 1, meta: { web_stories_tracking_optin: false } },
    isUpdating: false,
  };
}

function toggleOptInTracking(currentUser) {
  return {
    ...currentUser,
    data: {
      ...currentUser.data,
      meta: {
        web_stories_tracking_optin: !currentUser.data.meta
          .web_stories_tracking_optin,
      },
    },
  };
}
