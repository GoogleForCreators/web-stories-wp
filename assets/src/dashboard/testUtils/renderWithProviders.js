/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { FlagsProvider } from 'flagged';

/**
 * Internal dependencies
 */
import {
  theme as externalDesignSystemTheme,
  lightMode,
  SnackbarProvider,
} from '../../design-system';
import theme from '../theme';
import { ConfigProvider } from '../app/config';
import MockApiProvider from './mockApiProvider';

const defaultProviderValues = {
  features: {},
  theme: {
    DEPRECATED_THEME: theme,
    ...externalDesignSystemTheme,
    colors: lightMode,
  },
  config: {
    capabilities: { canReadPrivatePosts: true },
    allowedImageMimeTypes: ['image/png', 'image/jpeg', 'image/gif'],
  },
  api: {},
};

const DefaultWrapper = ({ children }) => children;

export const renderWithProviders = (
  ui,
  providerValues = {},
  renderOptions = {},
  wrapper = DefaultWrapper
) => {
  const mergedProviderValues = { ...defaultProviderValues, ...providerValues };

  const ProvidedWrapper = wrapper;

  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => (
    <ProvidedWrapper>
      <FlagsProvider features={mergedProviderValues.features}>
        <ThemeProvider theme={mergedProviderValues.theme}>
          <ConfigProvider config={mergedProviderValues.config}>
            <MockApiProvider value={mergedProviderValues.api}>
              <SnackbarProvider value={mergedProviderValues.snackbar}>
                {children}
              </SnackbarProvider>
            </MockApiProvider>
          </ConfigProvider>
        </ThemeProvider>
      </FlagsProvider>
    </ProvidedWrapper>
  );

  const mergedRenderOptions = { wrapper: Wrapper, ...renderOptions };

  return render(ui, mergedRenderOptions);
};

export default renderWithProviders;
