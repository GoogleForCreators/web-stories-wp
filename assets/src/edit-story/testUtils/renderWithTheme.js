/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { render, queries } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

/**
 * Internal dependencies
 */
import theme from '../theme';

import * as ariaLabelQueries from './queryByAriaLabel';
import * as autoAdvanceAfterQueries from './queryByAutoAdvanceAfter';
import * as idQueries from './queryById';

// eslint-disable-next-line react/prop-types
const WithThemeProvider = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const renderWithTheme = (ui, options) =>
  render(ui, {
    wrapper: WithThemeProvider,
    queries: {
      ...queries,
      ...ariaLabelQueries,
      ...autoAdvanceAfterQueries,
      ...idQueries,
    },
    ...options,
  });

export default renderWithTheme;
