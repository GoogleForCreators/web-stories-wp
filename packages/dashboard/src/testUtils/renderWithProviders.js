/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { FlagsProvider } from 'flagged';
import {
  theme as externalDesignSystemTheme,
  lightMode,
  SnackbarProvider,
  noop,
} from '@googleforcreators/design-system';

/**
 * Internal dependencies
 */
import { ConfigProvider } from '../app/config';
import MockApiProvider from './mockApiProvider';

const defaultProviderValues = {
  features: {},
  theme: {
    ...externalDesignSystemTheme,
    colors: lightMode,
  },
  config: {
    userId: 8675301,
    apiCallbacks: {
      createStoryFromTemplate: noop, // Not required in general, but without this "use template" button will not render and a test will fail.
    },
    styleConstants: {
      topOffset: 0,
    },
  },
  api: {},
};

const DefaultWrapper = ({ children }) => children;

export const renderWithProviders = (
  ui,
  providerValues = {},
  renderOptions = {},
  wrapper = DefaultWrapper
) => {
  const mergedProviderValues = { ...defaultProviderValues, ...providerValues };

  const ProvidedWrapper = wrapper;

  const Wrapper = ({ children }) => (
    <ProvidedWrapper>
      <FlagsProvider features={mergedProviderValues.features}>
        <ThemeProvider theme={mergedProviderValues.theme}>
          <ConfigProvider config={mergedProviderValues.config}>
            <MockApiProvider value={mergedProviderValues.api}>
              <SnackbarProvider value={mergedProviderValues.snackbar}>
                {children}
              </SnackbarProvider>
            </MockApiProvider>
          </ConfigProvider>
        </ThemeProvider>
      </FlagsProvider>
    </ProvidedWrapper>
  );

  const mergedRenderOptions = { wrapper: Wrapper, ...renderOptions };

  return render(ui, mergedRenderOptions);
};

export default renderWithProviders;
