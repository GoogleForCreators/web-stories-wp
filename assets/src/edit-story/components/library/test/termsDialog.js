/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/*
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * External dependencies
 */
import { waitFor, fireEvent, act } from '@testing-library/react';

/**
 * Internal dependencies
 */
import TermsDialog from '../panes/media/media3p/termsDialog';
import { renderWithTheme } from '../../../testUtils';

jest.mock('../../../utils/localStore', () => {
  const actual = jest.requireActual('../../../utils/localStore');
  return {
    ...actual,
    setItemByKey: jest.fn(),
    getItemByKey: jest.fn(() => false),
  };
});
import localStore, { LOCAL_STORAGE_PREFIX } from '../../../utils/localStore';

describe('TermsDialog', () => {
  it('should render', () => {
    const { queryByText } = renderWithTheme(<TermsDialog />);
    const link = queryByText(/^Learn more/);
    expect(link).toMatchSnapshot();
    expect(localStore.getItemByKey).toHaveBeenCalledWith(
      LOCAL_STORAGE_PREFIX.TERMS_MEDIA3P
    );
  });
  it('should store the acknowledgement in the localStore when the window is dismissed', async () => {
    const { queryByText } = renderWithTheme(<TermsDialog />);
    const dismiss = queryByText(/^Dismiss/);
    expect(localStore.getItemByKey).toHaveBeenCalledWith(
      LOCAL_STORAGE_PREFIX.TERMS_MEDIA3P
    );
    act(() => {
      fireEvent.click(dismiss);
    });
    await waitFor(() =>
      expect(localStore.setItemByKey).toHaveBeenCalledWith(
        LOCAL_STORAGE_PREFIX.TERMS_MEDIA3P,
        true
      )
    );
  });
});
