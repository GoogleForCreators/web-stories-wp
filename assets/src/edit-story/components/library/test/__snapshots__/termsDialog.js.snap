// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsDialog should render 1`] = `
<a
  aria-label="Learn more by visiting Web Stories for WordPress"
  href="https://wp.stories.google/docs#Terms"
  rel="noreferrer"
  target="_blank"
>
  Learn more.
</a>
`;
